src/server/plugins/engine/pageControllers/RepeatPageController.ts

```diff
@@ -179,6 +177,12 @@ export class RepeatPageController extends PageController {
       request: FormRequest,
       h: ResponseToolkit<FormRequestRefs>
     ) => {
+      if (!request.params.itemId) {
+        return h
+          .redirect(`${request.path}/${randomUUID()}${request.url.search}`)
+          .code(StatusCodes.SEE_OTHER)
+      }
+
```

Comment:
Thanks for adding the tests @davidjamesstone, coverage has jumped up now 🙌

Have you noticed we don't have coverage when `getRepeatAppData()` throws?

Wonder if that would be a good place to move this redirect? Instead of throwing

```js
throw badRequest('No repeat data found on the request')
```

Merge away, not a blocker

Reply:
`throw badRequest('No repeat data found on the request')` is only there for typescript narrowing purposes. `request.app.repeat` should never be undefined in `getAppRepeatData` (assuming `setAppRepeatData` has been called prior).

To me the top of `makeGetRouteHandler` is the natural place for the redirect to occur and while `request.app.repeat` should never be undefined, if it ever is and we change to do the redirect in that block, there's an increased risk of a redirect loop.